File: cli-tools/automation/node/model-crop.js
const axios = require('axios');
const { BASE_URL, getAccessToken } = require('./utils');

async function modelCrop() {
    const imageUrl = process.argv[2];
    const topCrop = process.argv[3] || null;
    const bottomCrop = process.argv[4] || null;

    if (!imageUrl) {
        console.log('Usage: node model-crop.js <imageUrl> [topCrop] [bottomCrop]');
        process.exit(1);
    }

    const token = await getAccessToken();
    try {
        const payload = { image_url: imageUrl };
        if (topCrop) payload.top_crop_location = topCrop;
        if (bottomCrop) payload.bottom_crop_location = bottomCrop;

        const response = await axios.post(`${BASE_URL}/images/model-crop`, payload, {
            headers: { 'Authorization': `Bearer ${token}` }
        });
        console.log(JSON.stringify(response.data, null, 2));
    } catch (error) {
        console.error('Error:', error.response ? error.response.data : error.message);
    }
}

modelCrop();

File: cli-tools/automation/python/model_crop.py
import sys
import json
import requests
from utils import BASE_URL, get_access_token

def model_crop():
    if len(sys.argv) < 2:
        print('Usage: python model_crop.py <imageUrl> [topCrop] [bottomCrop]')
        sys.exit(1)

    image_url = sys.argv[1]
    top_crop = sys.argv[2] if len(sys.argv) > 2 else None
    bottom_crop = sys.argv[3] if len(sys.argv) > 3 else None

    token = get_access_token()
    headers = {'Authorization': f'Bearer {token}', 'Content-Type': 'application/json'}
    payload = {'image_url': image_url}
    if top_crop: payload['top_crop_location'] = top_crop
    if bottom_crop: payload['bottom_crop_location'] = bottom_crop
    
    try:
        response = requests.post(f"{BASE_URL}/images/model-crop", json=payload, headers=headers)
        response.raise_for_status()
        print(json.dumps(response.json(), indent=2))
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    model_crop()
